services/src/pro/datasets/storage.rs

```diff
@@ -0,0 +1,197 @@
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+pub struct ChangeDatasetExpiration {
+    pub dataset_id: DatasetId,
+    pub expiration_change: ExpirationChange,
+}
+
+impl ChangeDatasetExpiration {
+    pub fn delete_meta(dataset_id: DatasetId) -> Self {
+        ChangeDatasetExpiration {
+            dataset_id,
+            expiration_change: ExpirationChange::SetExpire(Expiration {
+                deletion_timestamp: None,
+                delete_record: true,
+                delete_data: false,
+            }),
+        }
+    }
+
+    pub fn delete_fair(dataset_id: DatasetId) -> Self {
+        ChangeDatasetExpiration {
+            dataset_id,
+            expiration_change: ExpirationChange::SetExpire(Expiration {
+                deletion_timestamp: None,
+                delete_record: false,
+                delete_data: true,
+            }),
+        }
+    }
+
+    pub fn delete_full(dataset_id: DatasetId) -> Self {
+        ChangeDatasetExpiration {
+            dataset_id,
+            expiration_change: ExpirationChange::SetExpire(Expiration {
+                deletion_timestamp: None,
+                delete_record: true,
+                delete_data: true,
+            }),
+        }
+    }
+
+    pub fn delete_none(dataset_id: DatasetId) -> Self {
+        ChangeDatasetExpiration {
+            dataset_id,
+            expiration_change: ExpirationChange::SetExpire(Expiration {
+                deletion_timestamp: None,
+                delete_record: false,
+                delete_data: false,
+            }),
+        }
+    }
+
+    pub fn expire_meta(dataset_id: DatasetId, timestamp: DateTime) -> Self {
+        ChangeDatasetExpiration {
+            dataset_id,
+            expiration_change: ExpirationChange::SetExpire(Expiration {
+                deletion_timestamp: Some(timestamp),
+                delete_record: true,
+                delete_data: false,
+            }),
+        }
+    }
+
+    pub fn expire_fair(dataset_id: DatasetId, timestamp: DateTime) -> Self {
+        ChangeDatasetExpiration {
+            dataset_id,
+            expiration_change: ExpirationChange::SetExpire(Expiration {
+                deletion_timestamp: Some(timestamp),
+                delete_record: false,
+                delete_data: true,
+            }),
+        }
+    }
+
+    pub fn expire_full(dataset_id: DatasetId, timestamp: DateTime) -> Self {
+        ChangeDatasetExpiration {
+            dataset_id,
+            expiration_change: ExpirationChange::SetExpire(Expiration {
+                deletion_timestamp: Some(timestamp),
+                delete_record: true,
+                delete_data: true,
+            }),
+        }
+    }
+
+    pub fn expire_none(dataset_id: DatasetId, timestamp: DateTime) -> Self {
+        ChangeDatasetExpiration {
+            dataset_id,
+            expiration_change: ExpirationChange::SetExpire(Expiration {
+                deletion_timestamp: Some(timestamp),
```

Comment:
does it make sense to expire but not delete anything? What is the timestamp needed for?